packages/analytics-js/__tests__/components/configManager/ConfigManager.test.ts

```diff
@@ -0,0 +1,208 @@
+import { defaultHttpClient } from '@rudderstack/analytics-js/services/HttpClient';
+import { defaultErrorHandler } from '@rudderstack/analytics-js/services/ErrorHandler';
+import { defaultLogger } from '@rudderstack/analytics-js/services/Logger';
+import { ConfigManager } from '@rudderstack/analytics-js/components/configManager';
+import { state } from '@rudderstack/analytics-js/state';
+import { getSDKUrlInfo } from '@rudderstack/analytics-js/components/configManager/util/commonUtil';
+import { rest } from 'msw';
+import { server } from '../../../__mocks__/msw.server';
+import { dummySourceConfigResponse } from '../../../__mocks__/fixtures';
+
+jest.mock('../../../src/services/Logger', () => {
+  const originalModule = jest.requireActual('../../../src/services/Logger');
+
+  return {
+    __esModule: true,
+    ...originalModule,
+    defaultLogger: {
+      error: jest.fn((): void => {}),
+    },
+  };
+});
+
+jest.mock('../../../src/services/ErrorHandler', () => {
+  const originalModule = jest.requireActual('../../../src/services/ErrorHandler');
+
+  return {
+    __esModule: true,
+    ...originalModule,
+    defaultErrorHandler: {
+      onError: jest.fn((): void => {}),
+    },
+  };
+});
+
+jest.mock('../../../src/components/configManager/util/commonUtil.ts', () => {
+  const originalModule = jest.requireActual(
+    '../../../src/components/configManager/util/commonUtil.ts',
+  );
+
+  return {
+    __esModule: true,
+    ...originalModule,
+    getSDKUrlInfo: jest.fn(),
+  };
+});
+
+describe('ConfigManager', () => {
+  let configManagerInstance: ConfigManager;
+  const errorMsg = 'Unable to load the SDK due to invalid write key: "INVALID-WRITE-KEY"';
+  const errorMsgSourceConfigResponse = 'Unable to fetch source config';
+  const sampleWriteKey = '2LoR1TbVG2bcISXvy7DamldfkgO';
+  const sampleDataPlaneUrl = 'https://www.dummy.url';
+  const sampleDestSDKUrl = 'https://www.sample.url/integrations';
+  const sampleConfigUrl = 'https://dummy.dataplane.host.com';
+  const sampleScriptURL = 'https://www.dummy.url/fromScript/v3/rudder-analytics.min.js';
+  const lockIntegrationsVersion = false;
+
+  beforeAll(() => {
+    server.listen();
+  });
+
+  beforeEach(() => {
+    configManagerInstance = new ConfigManager(
+      defaultHttpClient,
+      defaultErrorHandler,
+      defaultLogger,
+    );
+  });
+
+  afterEach(() => {
+    server.resetHandlers();
+    server.events.removeAllListeners();
+  });
+
+  afterAll(() => {
+    server.close();
+  });
+
+  it('should throw an error for invalid writeKey', () => {
+    state.lifecycle.writeKey.value = 'INVALID-WRITE-KEY';
+    try {
+      configManagerInstance.init();
+    } catch (e) {
+      expect(e.message).toBe(errorMsg);
+    }
```

Comment:
The correct way to test exception-throwing code.
```suggestion
    expect(() => { configManagerInstance.init() }).toThrow(errorMsg);
```